iroh/src/magicsock/transports/relay/actor.rs

```diff
@@ -498,7 +498,10 @@ impl ActiveRelayActor {
             match time::timeout(CONNECT_TIMEOUT, client_builder.connect()).await {
                 Ok(Ok(client)) => Ok(client),
                 Ok(Err(err)) => Err(ConnectSnafu.into_error(err)),
-                Err(_) => Err(TimeoutSnafu.build()),
+                Err(_) => Err(TimeoutSnafu {
+                    timeout: CONNECT_TIMEOUT,
+                }
+                .build()),
```

Comment:
You can also do
```suggestion
                Err(_) => TimeoutSnafu {
                    timeout: CONNECT_TIMEOUT,
                }.fail(),
```
(Same goes for the other branches actually.

Reply:
Oh, neat.  But I'm going to leave the style alone for now.